### ISSUE TITLE
cropped image url always tries to crop from large image

### ISSUE THREAD

**Issue by 1aurabrown** [artsy/metaphysics#82]
https://github.com/artsy/metaphysics/blob/master/schema/image/cropped.js#L11-L14

This should be able to determine the largest available version and crop that one. Lots of profile cover images for the partner page don't have a `large` version, and so trying to request the cropped url from metaphysics results in a nonexistent image url, like 
```
background-image: url(https://i.embed.ly/1/display/crop?url=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2F0605qKOomA-mNhx61j9vDA%2Flarge.jpg&width=400&height=300&key=a1f82558d8134f6cbebceb9e67d04980&quality=95);
```
see the actual profile data:
https://api.artsy.net/api/v1/profile/kristina-parsons

**Comment by dzucconi** [comment]
For now you can just specify the version you want with `version: foo`

But yeah; I think we should just not set a default anymore and force the consumer to specify which version they want by making that argument non-null.

I don't like the idea of trying to set a default because then we have to maintain all knowledge of versions and their sizes here. This would be fine if it was even remotely sane. [It, however, is not.](https://docs.google.com/spreadsheets/u/1/d/1mmfQ2NfmK2hVxXHSEvgAQwnnybKy0qs2blNx5VYAELE/edit#gid=0)

**Comment by 1aurabrown** [comment]
right but before the request happens, i can't possibly know what versions are available

**Comment by dzucconi** [comment]
I think presumably the idea is that you'd peek at what the versions were while crafting your query
```
{
  profile(id: "kristina-parsons") {
    id
    image {
      id
      versions
    }
  }
}
```

Then use whichever is appropriate.

Though I realize now I fixed this behavior in https://github.com/artsy/metaphysics/pull/80 earlier today.

So this does the right thing now by preventing you from using a version that doesn't exist: 

https://github.com/artsy/metaphysics/blob/7f1e3b1f0abd105902479aa34130c9f8730ed450/schema/image/normalize.js#L16-L26

we just haven't promoted it to production yet.

It's still a bit weird because you can specify `large` knowing it doesn't exist and it will not error, it will just silently convert that to `wide`. It has been the case where images will be missing one or two specific versions seemingly at random, so throwing a real error probably isn't a great idea either.

**Comment by alloy** [comment]
@1aurabrown Would this have been the better fix for https://github.com/artsy/force/pull/418 ?

**Comment by alloy** [comment]
@1aurabrown bump

**Comment by 1aurabrown** [comment]
@alloy I'm not sure I totally understand. The issue with the tall artworks is that the image version being used was not sufficiently large for the space the image needed to occupy, but for the vast majority of artworks is bigger than necessary. Since my query is fetching the artwork's url for a certain version, not sure how to look at the available versions first without making two subsequent queries -- one to get the versions and aspect ratio, another for the url (or alternatively we could just do the string interpolation for the image url on the client)

**Comment by 1aurabrown** [comment]
I would say, for that issue, perhaps we should do as joey suggested -- maybe we can make an image version that, no matter what the aspect ratio, is always the same width. a width that is wide enough to accommodate our three-column view without up-sampling. That way we won't won't have pixellated images, nor would we have to request an unnecessarily large image version for all images that aren't skinny, just for the sake of accommodating the skinny ones, nor would we have to do any kind of logic on the client to take an educated guess at a variable version as a function of the aspect ratio.

**Comment by alloy** [comment]
Yeah that does sound like a good initial step.
 I actually don't like at all that the client has to pass in a versions besides just the requested crop/fit size.

**Comment by 1aurabrown** [comment]
i agree that if you are resizing, the smallest sufficient source image should be used. i'm not sure how cropping works exactly, because the area cropped is independent of how big the final image would be (for example, i could crop the upper right quadrant of an image, at various sizes, from icon sized to full-res)

in this case, if it's fine to just used a standard image version instead of a resize, that seems best.

**Comment by alloy** [comment]
> because the area cropped is independent of how big the final image would be (for example, i could crop the upper right quadrant of an image, at various sizes, from icon sized to full-res)

Does Gemini actually support that _and_ do we actually use that anywhere?
